src/main/java/net/sf/jabref/gui/BasePanel.java

```diff
@@ -709,9 +711,22 @@ public void update() {
     private void copyCitationToClipboard(CitationStyleOutputFormat outputFormat) {
         EventList<BibEntry> selectedEntries = mainTable.getSelected();
         String style = citationStyleCache.getCitationStyle().getSource();
+        new SwingWorker<String, Void>() {
+            @Override
+            protected String doInBackground() throws Exception {
+                return CitationStyleGenerator.generateCitations(selectedEntries, style, outputFormat);
+            }
 
-        String citation = CitationStyleGenerator.generateCitations(selectedEntries, style, outputFormat);
-        new ClipBoardManager().setClipboardContents(citation);
+            @Override
+            public void done() {
+                try {
+                    new ClipBoardManager().setClipboardContents(get());
+                    frame.setStatus(Localization.lang("Copied citations to the clipboard!"));
```

Comment:
Please output the number. Use `%0`. Use `.` as final sign, not `!`.

> Copied %0 citations.

I think, "the clipboard" is obsolete.